File: src/main/scala/com/github/kazuhito_m/twitterbattler/primitive/domain/model/battle/Turn.scala
package com.github.kazuhito_m.twitterbattler.primitive.domain.model.battle

import com.github.kazuhito_m.twitterbattler.primitive.domain.model.battle.activitiy.Activities
import com.github.kazuhito_m.twitterbattler.primitive.domain.model.party.PartyStatus

class Turn(
            val beforeStatus: PartyStatus,
            val afterStatus: PartyStatus,
            val activities: Activities
          ) {

  def this(status: PartyStatus) = {
    this(status, status, new Activities)
  }

}

File: server/src/main/java/com/github/kazuhito_m/twitterbattler/primitive/domain/model/battle/Turn.java
package com.github.kazuhito_m.twitterbattler.primitive.domain.model.battle;

import com.github.kazuhito_m.twitterbattler.primitive.domain.model.battle.activitiy.Activities;
import com.github.kazuhito_m.twitterbattler.primitive.domain.model.party.PartyStatus;

public class Turn {
    final PartyStatus beforeStatus;
    final PartyStatus afterStatus;
    final Activities activities;

    public Turn(PartyStatus beforeStatus, PartyStatus afterStatus, Activities activities) {
        this.beforeStatus = beforeStatus;
        this.afterStatus = afterStatus;
        this.activities = activities;
    }
}
